fern/docs/pages/openapi/extensions.mdx

```diff
@@ -30,6 +30,54 @@ paths:
       x-fern-sdk-method-name: create_user
 ```
 
+## Authentication overrides
+
+For authentication within your SDK, you may want to customize the:
+- Names of the parameters a user will populate to provide their credentials
+- Any environment variable the SDK can scan for to automatically provide the user's credentials automatically
+
+By default, Fern uses selects names for authentication parameters like api_key, username and password, etc.
+Note that this configuration is optional, and even when provided `name` or `envvar` may be provided alone, both are not required.
+
+For bearer authentication:
+
+```yaml openapi.yaml
+x-fern-bearer:
+  name: apiKey
+  envvar: FERN_API_KEY
+```
+
+In a Python SDK for example, the above configuration would produce a client that looks like:
+
+```python
+import os
+
+class Client:
+
+  def __init__(self, *, apiKey: str = os.getenv("FERN_API_KEY"))
+```
+
+Similarly, for basic authentication the configuration is:
+
+```yaml openapi.yaml
+x-fern-basic:
+  username:
+     name: username
+     envvar: MY_USERNAME
+  password:
+     name: password
+     envvar: MY_PASSWORD
+```
+
+And for a custom auth header configuration:
+
+```yaml openapi.yaml
+x-fern-header:
+  name: header
+  envvar: MY_HEADER_ENVVAR
+```
```

Comment:
@armandobelardo should we make it clear that this lives inside of the security scheme defined in 
```
  components: 
    securitySchemes: 
       A: 
        x-fern-bearer: 
           ....
```

Reply:
It doesn't live in security schemes, it's top level, my vscode was giving me a bunch of red lines in the yml when it was under security schemes so I had made it top level
I can move it back and update the docs

Reply:
The only issue with having it top level is that folks can have multiple headers / multiple schemes (and while we dont support that now) we'll probably want to support it at some point in the future

Reply:
In the fern definition we also just call it `env` instead of `envvar` -- thoughts on adopting that here?

Reply:
Updated both!